neutron/plugins/ml2/drivers/openvswitch/agent/ovs_neutron_agent.py

```diff
@@ -1654,10 +1654,23 @@ def process_network_ports(self, port_info, ovs_restarted):
         self._add_port_tag_info(need_binding_devices)
         if security_disabled_ports:
             added_ports -= set(security_disabled_ports)
-        self.sg_agent.setup_port_filters(added_ports,
-                                         port_info.get('updated', set()))
+
+        # The following call was split into two calls so that added_ports
+        # can be bound quicker.
+        #self.sg_agent.setup_port_filters(added_ports,
+        #                                 port_info.get('updated', set()))
+
+        devices_to_refilter, global_refresh_firewall = (
+            self.sg_agent.setup_port_filters_part_1(added_ports))
+
         failed_devices['added'] |= self._bind_devices(need_binding_devices)
```

Comment:
Nate, are you able to run a performance test with this patch and your iptables patch to determine the potential performance gain that this patch is trying to achieve?